middleware/context_timeout.go

```diff
@@ -0,0 +1,89 @@
+package middleware
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/labstack/echo/v4"
+)
+
+// ContextTimeoutConfig defines the config for ContextTimeout middleware.
+type ContextTimeoutConfig struct {
+	// Skipper defines a function to skip middleware.
+	Skipper Skipper
+
+	// ErrorHandler is a function when error aries in middeware execution.
+	ErrorHandler func(err error, c echo.Context) error
+
+	// Timeout configures a timeout for the middleware, defaults to 0 for no timeout
+	Timeout time.Duration
+}
+
+var (
+	// DefaultContextTimeoutErrorHandler is default error handler of ContextTimeout middleware.
+	DefaultContextTimeoutErrorHandler = func(err error, c echo.Context) error {
```

Comment:
this is not needed anymore as `ToMiddleware()` has it already. These package public things are sometimes pain as anyone would mutate them.